src/app/CommandSender.cpp

```diff
@@ -330,5 +355,62 @@ TLV::TLVWriter * CommandSender::GetCommandDataIBTLVWriter()
     }
 }
 
+CHIP_ERROR CommandSender::SendTimedRequest(uint16_t aTimeoutMs)
+{
+    using namespace Protocols::InteractionModel;
+    using namespace Messaging;
+
+    // The payload is an anonymous struct (2 bytes) containing a single
+    // 16-bit integer with a context tag (1 control byte, 1 byte tag, at
+    // most 2 bytes for the integer).  Use MessagePacketBuffer::New to
+    // account for other message-global overheads (MIC, etc).
+    System::PacketBufferHandle payload = MessagePacketBuffer::New(6);
+    VerifyOrReturnError(!payload.IsNull(), CHIP_ERROR_NO_MEMORY);
+
+    System::PacketBufferTLVWriter writer;
+    writer.Init(std::move(payload));
+
+    TimedRequestMessage::Builder builder;
+    ReturnErrorOnFailure(builder.Init(&writer));
+
+    builder.TimeoutMs(aTimeoutMs);
+    ReturnErrorOnFailure(builder.GetError());
+
+    ReturnErrorOnFailure(writer.Finalize(&payload));
+
+    ReturnErrorOnFailure(mpExchangeCtx->SendMessage(MsgType::TimedRequest, std::move(payload), SendMessageFlags::kExpectResponse));
```

Comment:
Do this check VerifyOrReturnError(apExchangeContext != nullptr, CHIP_ERROR_INCORRECT_STATE);?

Reply:
I guess we could, but this function has precisely one caller and that caller definitely has a non-null exchange....